sdk/core/core/src/az_span.c

```diff
@@ -129,6 +129,82 @@ AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_find(az_span span, az_span target, az_span* out_span)
+{
+  AZ_PRECONDITION_VALID_SPAN(span, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(target, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
+  uint8_t* span_ptr = az_span_ptr(span);
+  int32_t span_length = az_span_length(span);
+  uint8_t* target_ptr = az_span_ptr(target);
+  int32_t target_length = az_span_length(target);
+
+  for (int i = 0; i < span_length; i++)
+  {
+    if (span_ptr[i] == target_ptr[0])
+    {
+      int j;
+      for (j = 1; j < target_length; j++)
+      {
+        if (span_ptr[i + j] != target_ptr[j])
+        {
+            break;
+        }
+      }
+
+      if (j == target_length)
+      {
+        out_span->_internal.ptr = span_ptr + i;
+        out_span->_internal.length = target_length;
+        out_span->_internal.capacity = target_length;
+
+        return AZ_OK;
+      }
+    }
+  }
+
+  return AZ_ERROR_EOF;
+}
+
+AZ_NODISCARD az_result az_span_token_next(az_span span, az_span delim, az_span* token, az_span* out_span)
+{
+  AZ_PRECONDITION_VALID_SPAN(span, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(delim, 1, false);
+  AZ_PRECONDITION_NOT_NULL(token);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
+  az_span instance;
+
+  token->_internal.ptr = span._internal.ptr;
```

Comment:
We prefer calling az_span_init in the if & else; this makes he code more maintainable and can catch some programmer errors.

Reply:
token is an out parameter and so should be named out_token then.

Reply:
Done.